Unable to deploy app: index out of range [1] with length 1 GetContainerPort

## Description of problem
After a fresh installation of Dokku on Ubuntu Server 18.04, I am unable to deploy an app from a docker image. I get the following error:

panic: runtime error: index out of range [1] with length 1

### How reproducible

Tried using the command git:from-image and also with a git push from my local machine:

dokku git:from-image vpn kylemanna/openvpn:latest

### Steps to Reproduce

1. dokku git:from-image vpn kylemanna/openvpn:latest

#### Actual Results
```
-----> Deploying vpn...
 !     No procfile found
2021/09/27 13:17:56 exit status 1
panic: runtime error: index out of range [1] with length 1

goroutine 1 [running]:
github.com/dokku/dokku/plugins/network.GetContainerPort(0x7fff513af22a, 0x3, 0x7fff513af22e, 0x3, 0x7fff513af232, 0x40, 0x4d3f00, 0xc000054180, 0x0)
	/go/src/github.com/dokku/dokku/plugins/network/network.go:181 +0x5be
github.com/dokku/dokku/plugins/network.TriggerNetworkGetPort(0x7fff513af22a, 0x3, 0x7fff513af22e, 0x3, 0x7fff513af232, 0x40, 0xc0000a2000, 0x8, 0x8)
	/go/src/github.com/dokku/dokku/plugins/network/triggers.go:119 +0x76
main.main()
	/go/src/github.com/dokku/dokku/plugins/network/src/triggers/triggers.go:57 +0x478
2021/09/27 13:17:58 exit status 2
2021/09/27 13:17:58 exit status 1
2021/09/27 13:17:58 exit status 1
2021/09/27 13:17:58 exit status 1
```

#### Expected Results

Succesful deployment of the app

## Environment Information

### `dokku report vpn` output

```
-----> uname: Linux chavito 4.15.0-158-generic #166-Ubuntu SMP Fri Sep 17 19:37:52 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
-----> memory: 
                     total        used        free      shared  buff/cache   available
       Mem:            985         224         204           8         557         609
       Swap:           951           1         950
-----> docker version: 
       Client: Docker Engine - Community
        Version:           20.10.8
        API version:       1.41
        Go version:        go1.16.6
        Git commit:        3967b7d
        Built:             Fri Jul 30 19:54:08 2021
        OS/Arch:           linux/amd64
        Context:           default
        Experimental:      true
       
       Server: Docker Engine - Community
        Engine:
         Version:          20.10.8
         API version:      1.41 (minimum version 1.12)
         Go version:       go1.16.6
         Git commit:       75249d8
         Built:            Fri Jul 30 19:52:16 2021
         OS/Arch:          linux/amd64
         Experimental:     false
        containerd:
         Version:          1.4.9
         GitCommit:        e25210fe30a0a703442421b0f60afac609f950a3
        runc:
         Version:          1.0.1
         GitCommit:        v1.0.1-0-g4144b63
        docker-init:
         Version:          0.19.0
         GitCommit:        de40ad0
-----> docker daemon info: 
WARNING: No swap limit support
       Client:
        Context:    default
        Debug Mode: true
        Plugins:
         app: Docker App (Docker Inc., v0.9.1-beta3)
         buildx: Build with BuildKit (Docker Inc., v0.6.1-docker)
         scan: Docker Scan (Docker Inc., v0.8.0)
       
       Server:
        Containers: 6
         Running: 0
         Paused: 0
         Stopped: 6
        Images: 18
        Server Version: 20.10.8
        Storage Driver: overlay2
         Backing Filesystem: extfs
         Supports d_type: true
         Native Overlay Diff: true
         userxattr: false
        Logging Driver: json-file
        Cgroup Driver: cgroupfs
        Cgroup Version: 1
        Plugins:
         Volume: local
         Network: bridge host ipvlan macvlan null overlay
         Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
        Swarm: inactive
        Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
        Default Runtime: runc
        Init Binary: docker-init
        containerd version: e25210fe30a0a703442421b0f60afac609f950a3
        runc version: v1.0.1-0-g4144b63
        init version: de40ad0
        Security Options:
         apparmor
         seccomp
          Profile: default
        Kernel Version: 4.15.0-158-generic
        Operating System: Ubuntu 18.04.1 LTS
        OSType: linux
        Architecture: x86_64
        CPUs: 1
        Total Memory: 985.6MiB
        Name: chavito
        ID: BVOX:SQ3G:ZXUP:4FYX:REOE:6GRI:VSHG:EW26:OBIW:GXD5:DUMG:ATCU
        Docker Root Dir: /var/lib/docker
        Debug Mode: false
        Registry: https://index.docker.io/v1/
        Labels:
        Experimental: false
        Insecure Registries:
         127.0.0.0/8
        Live Restore Enabled: false
       
-----> git version: git version 2.17.1
-----> sigil version: 0.6.0
-----> herokuish version: 
       herokuish: 0.5.30
       buildpacks:
         heroku-buildpack-multi     v1.0.0
         heroku-buildpack-ruby      v228
         heroku-buildpack-nodejs    v185
         heroku-buildpack-clojure   v87
         heroku-buildpack-python    v197
         heroku-buildpack-java      v69
         heroku-buildpack-gradle    v35
         heroku-buildpack-scala     v90
         heroku-buildpack-play      v26
         heroku-buildpack-php       v196
         heroku-buildpack-go        v153
         buildpack-nginx            v14
         buildpack-null             v3
-----> dokku version: dokku version 0.25.4
-----> plugn version: plugn: 0.6.1
-----> dokku plugins: 
         00_dokku-standard    0.25.4 enabled    dokku core standard plugin
         20_events            0.25.4 enabled    dokku core events logging plugin
         app-json             0.25.4 enabled    dokku core app-json plugin
         apps                 0.25.4 enabled    dokku core apps plugin
         builder              0.25.4 enabled    dokku core builder plugin
         builder-dockerfile   0.25.4 enabled    dokku core builder-dockerfile plugin
         builder-herokuish    0.25.4 enabled    dokku core builder-herokuish plugin
         builder-null         0.25.4 enabled    dokku core builder-null plugin
         builder-pack         0.25.4 enabled    dokku core builder-pack plugin
         buildpacks           0.25.4 enabled    dokku core buildpacks plugin
         certs                0.25.4 enabled    dokku core certificate management plugin
         checks               0.25.4 enabled    dokku core checks plugin
         common               0.25.4 enabled    dokku core common plugin
         config               0.25.4 enabled    dokku core config plugin
         cron                 0.25.4 enabled    dokku core cron plugin
         docker-options       0.25.4 enabled    dokku core docker-options plugin
         domains              0.25.4 enabled    dokku core domains plugin
         enter                0.25.4 enabled    dokku core enter plugin
         git                  0.25.4 enabled    dokku core git plugin
         logs                 0.25.4 enabled    dokku core logs plugin
         network              0.25.4 enabled    dokku core network plugin
         nginx-vhosts         0.25.4 enabled    dokku core nginx-vhosts plugin
         plugin               0.25.4 enabled    dokku core plugin plugin
         proxy                0.25.4 enabled    dokku core proxy plugin
         ps                   0.25.4 enabled    dokku core ps plugin
         registry             0.25.4 enabled    dokku core registry plugin
         repo                 0.25.4 enabled    dokku core repo plugin
         resource             0.25.4 enabled    dokku core resource plugin
         run                  0.25.4 enabled    dokku core run plugin
         scheduler-docker-local 0.25.4 enabled    dokku core scheduler-docker-local plugin
         scheduler-null       0.25.4 enabled    dokku core scheduler-null plugin
         shell                0.25.4 enabled    dokku core shell plugin
         ssh-keys             0.25.4 enabled    dokku core ssh-keys plugin
         storage              0.25.4 enabled    dokku core storage plugin
         tags                 0.25.4 enabled    dokku core tags plugin
         tar                  0.25.4 enabled    dokku core tar plugin
         trace                0.25.4 enabled    dokku core trace plugin
=====> vpn app-json information
       App json computed selected:    app.json
       App json global selected:      app.json
       App json selected:             
=====> vpn app information
       App deploy source:             
       App dir:                       /home/dokku/vpn
       App locked:                    true
=====> vpn builder information
       Builder build dir:             
       Builder computed build dir:    
       Builder computed selected:     
       Builder global build dir:      
       Builder global selected:       
       Builder selected:              
=====> vpn builder-dockerfile information
       Builder dockerfile computed dockerfile path: Dockerfile               
       Builder dockerfile global dockerfile path: Dockerfile               
       Builder dockerfile dockerfile path:                          
=====> vpn builder-pack information
       Builder pack computed projecttoml path: project.toml             
       Builder pack global projecttoml path: project.toml             
       Builder pack projecttoml path:                          
=====> vpn buildpacks information
       Buildpacks computed stack:     gliderlabs/herokuish:latest-20
       Buildpacks global stack:       
       Buildpacks list:               
       Buildpacks stack:              
=====> vpn ssl information
       Ssl dir:                       /home/dokku/vpn/tls      
       Ssl enabled:                   false                    
       Ssl hostnames:                                          
       Ssl expires at:                                         
       Ssl issuer:                                             
       Ssl starts at:                                          
       Ssl subject:                                            
       Ssl verified:                                           
=====> vpn checks information
       Checks disabled list:          none                     
       Checks skipped list:           none                     
=====> vpn cron information
       Cron task count:               0
=====> vpn docker options information
       Docker options build:                                   
       Docker options deploy:         --restart=on-failure:10 -v /var/lib/dokku/data/storage/vpn:/etc/openvpn 
       Docker options run:            -v /var/lib/dokku/data/storage/vpn:/etc/openvpn 
=====> vpn domains information
       Domains app enabled:           true                     
       Domains app vhosts:                                     
       Domains global enabled:        true                     
       Domains global vhosts:         moonfishteam.com         
=====> vpn git information
       Git deploy branch:             master                   
       Git global deploy branch:      master                   
       Git keep git dir:              false                    
       Git rev env var:               GIT_REV                  
       Git sha:                       f82fc8b                  
       Git last updated at:           1632741450               
=====> vpn logs information
       Logs computed max size:        10m
       Logs global max size:          10m
       Logs global vector sink:       
       Logs max size:                 
       Logs vector sink:              
=====> vpn network information
       Network attach post create:           
       Network attach post deploy:           
       Network bind all interfaces:          false
       Network computed attach post create:  
       Network computed attach post deploy:  
       Network computed bind all interfaces: false
       Network computed initial network:     
       Network computed tld:                 
       Network global attach post create:    
       Network global attach post deploy:    
       Network global bind all interfaces:   false
       Network global initial network:       
       Network global tld:                   
       Network initial network:              
       Network static web listener:          
       Network tld:                          
       Network web listeners:                
=====> vpn nginx information
       Nginx access log format:                                
       Nginx access log path:         /var/log/nginx/vpn-access.log
       Nginx bind address ipv4:                                
       Nginx bind address ipv6:       ::                       
       Nginx client max body size:                             
       Nginx disable custom config:   false                    
       Nginx error log path:          /var/log/nginx/vpn-error.log
       Nginx global hsts:             true                     
       Nginx computed hsts:           true                     
       Nginx hsts:                                             
       Nginx hsts include subdomains: true                     
       Nginx hsts max age:            15724800                 
       Nginx hsts preload:            false                    
       Nginx proxy buffer size:       4096                     
       Nginx proxy buffering:         on                       
       Nginx proxy buffers:           8 4096                   
       Nginx proxy busy buffers size: 8192                     
       Nginx proxy read timeout:      60s                      
       Nginx last visited at:                                  
       Nginx x forwarded for value:   $remote_addr             
       Nginx x forwarded port value:  $server_port             
       Nginx x forwarded proto value: $scheme                  
       Nginx x forwarded ssl:                                  
=====> vpn proxy information
       Proxy enabled:                 true
       Proxy port map:                http:80:5000
       Proxy type:                    nginx
=====> vpn ps information
       Deployed:                      false
       Processes:                     0
       Ps can scale:                  true
       Ps computed procfile path:     Procfile
       Ps global procfile path:       Procfile
       Ps procfile path:              
       Ps restart policy:             on-failure:10
       Restore:                       true
       Running:                       false
=====> vpn registry information
       Registry computed image repo:      dokku/vpn
       Registry computed push on release: false
       Registry computed server:          
       Registry global push on release:   
       Registry global server:            
       Registry image repo:               
       Registry push on release:          
       Registry server:                   
       Registry tag version:              
=====> vpn resource information
=====> vpn scheduler-docker-local information
       Scheduler docker local disable chown:                          
=====> vpn storage information
       Storage build mounts:                                   
       Storage deploy mounts:         -v /var/lib/dokku/data/storage/vpn:/etc/openvpn 
       Storage run mounts:            -v /var/lib/dokku/data/storage/vpn:/etc/openvpn 
```
> This is required! Issues missing this information may be closed.
>
> For problems affecting all applications, the report output for a broken application is useful for our debugging.
> In these cases, you may run `dokku report` without any arguments to display the top-level reporting information.

### How (deb/make/rpm) and where (AWS, VirtualBox, physical, etc.) was Dokku installed?

Dokku was installed using curl | bash on a VPS.

### Additional information

- Container Inspect Output (if applicable) via `dokku ps:inspect APP_NAME`
```
dokku ps:inspect vpn
"docker container inspect" requires at least 1 argument.
See 'docker container inspect --help'.

Usage:  docker container inspect [OPTIONS] CONTAINER [CONTAINER...]

Display detailed information on one or more containers
Traceback (most recent call last):
  File "/tmp/dokku-6131-trigger-scheduler-docker-local-scheduler-inspect.taIbfS", line 5, in <module>
    data = json.load(sys.stdin)
  File "/usr/lib/python3.6/json/__init__.py", line 299, in load
    parse_constant=parse_constant, object_pairs_hook=object_pairs_hook, **kw)
  File "/usr/lib/python3.6/json/__init__.py", line 354, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3.6/json/decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python3.6/json/decoder.py", line 357, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
2021/09/27 13:31:28 exit status 1
 !     exit status 1
```

- The nginx configuration (if applicable) via `dokku nginx:show-config APP_NAME`

dokku nginx:show-config vpn
```
 !     No nginx.conf exists for vpn
```
- Link to the exact repository being deployed (if possible/applicable): 
Tried:
https://github.com/v-yarotsky/dokku-openvpn
https://hub.docker.com/r/kylemanna/openvpn

- If a deploy is failing or behaving unexpectedly:
  - Application name: dokku-openvpn

Thank you very much!

--- a/plugins/network/network.go
+++ b/plugins/network/network.go
@@ -174,11 +174,14 @@ func GetContainerPort(appName, processType string, containerID string, isHerokui
 				break
 			}
 		}
-		cmd := sh.Command(common.DockerBin(), "container", "port", containerID, port)
-		cmd.Stderr = ioutil.Discard
-		b, err := cmd.Output()
-		if err == nil {
-			port = strings.Split(string(b[:]), ":")[1]
+
+		if port != "" {
+			cmd := sh.Command(common.DockerBin(), "container", "port", containerID, port)
+			cmd.Stderr = ioutil.Discard
+			b, err := cmd.Output()
+			if err == nil {
+				port = strings.Split(string(b[:]), ":")[1]
+			}
 		}
 	} else {
 		port = "5000"

--- a/tests/unit/network.bats
+++ b/tests/unit/network.bats
@@ -241,3 +241,25 @@ teardown() {
   echo "status: $status"
   assert_success
 }
+
+@test "(network) handle single-udp app" {
+  run /bin/bash -c "dokku docker-options:add $TEST_APP -p 1194:1194"
+  echo "output: $output"
+  echo "status: $status"
+  assert_success
+
+  run /bin/bash -c "dokku checks:disable $TEST_APP"
+  echo "output: $output"
+  echo "status: $status"
+  assert_success
+
+  # The app _should_ fail to deploy because it requires some config file
+  # but we really only care about getting to pre-flight checks so that is fine.
+  # It does not because of some yet unknown bug in deploying when zero-downtime is disable...
+  run /bin/bash -c "dokku git:from-image $TEST_APP kylemanna/openvpn:latest"
+  echo "output: $output"
+  echo "status: $status"
+  assert_success
+  assert_output_contains "Attempting pre-flight checks" 0
+  assert_output_contains "Application deployed"
+}

